core/controllers/admin.py

```diff
@@ -382,7 +382,7 @@ def get(self):
                 'Exploration \'%s\' does not have \'%s\' state.'
                 % (exp_id, state_name))
 
-        state_answers = stats_services.get_state_answers(
+        state_answers = stats_services_old.get_state_answers(
```

Comment:
For each of these references to stats_services_old, can you add a `TODO(pranavsid98): ` saying this should be changed to stats_services.XXX (where you should specify what XXX is)?

Or is our convention going to be that we need to create isomorphic functions (like get_state_answers, etc.) in stats_services?  If so then maybe add empty functions for these there and have the TODOs there instead, e.g. `TODO(pranavsid98): Implement this function and update all references to stats_services_old.function_name to point to this one instead.`

Basically, you're introducing a temporary non-ideal state of affairs here, and we need to have complete info about what needs to be done in order to get to the desired end state, so that doing so is almost mechanical.

Reply:
Changed it all back to one stats_services file. So, no issue here.

Reply:
@seanlip We might just be emulating a lot of these functions in the future, so I'm inclined towards your second suggestion. The whole idea is: After we are done with the full implementation, we should be able to replace 'stats_services_old' with 'stats_services' globally and just remove stats_services_old.

Reply:
Do we have a clear plan for how we'll emulate them? The new functions that should be implemented should be listed in the design doc and we should have a clear idea of how they're going to work, otherwise we risk ending up with an indeterminate intermediate state.

Reply:
@seanlip We don't have a clear definition for the visualisations framework yet. This will require solid documentation on par with the statistics framework and only then, will we be able to decide which functions we might need. For now, I've listed the functions in stats_services_old that we will be emulating in stats_services. Logically, these seem to be the only functions that will be involved in the visualisations pipeline. Again, I'm not really sure if there are even any problems with the visualisations framework - which again needs research.
Do you suggest we chalk up the visualisations framework right now? Or lets just go forward with this rough idea of functions we might need. Since these functions are labelled with TODOs, we won't miss them (shouldn't be any worry of them ending up as dead code).

Reply:
In particular, how are you planning to emulate them so we don't have the dependency issue?

Reply:
@seanlip So, when we broke up the dependency between exp_services and stats_services, we ended up getting a transitive dependency via stats_job_continuous. We retrieved data for the statistics and visualisations from the continuously running aggregator job. According to the current stats framework, we aren't using an aggregator job. A similar approach to tackle the visualisations framework should leave our stats_service module without any dependency from stats_jobs_continuous. Hence, no dependency with exp_services.

Reply:
That sounds reasonable. Are there any other dependency issues, or is the stats_job_continuous one the last one?

The only other thing I would suggest is to incorporate these into the milestones. I.e. tasks for some of the milestones should be "Migrate stats_services_old.XXX to stats_services.XXX so that stats can be retrieved without using the aggregator job." We should migrate stuff to stats_services as soon as possible once the precondition is completed, and that can, in a way, be used as a success criterion for the milestone in question. Please ensure that all the functions in stats_services_old do get represented in the milestones doc.

Reply:
@seanlip Yep. I'll do that for sure. Whenever I get the time, I'll also start documenting the visualisations framework - if we manage to get that up before this project is done, it should give us a better picture of the stuff we need in stats_services.

Reply:
OK, thanks! Let's please do that before this gets merged, so that we have a deterministic future plan. I'll review the doc before approving this, so let me know when it's ready!